packages/keyring-controller/src/KeyringController.ts

```diff
@@ -1796,6 +1802,11 @@ export class KeyringController extends BaseController<
       `${name}:signUserOperation`,
       this.signUserOperation.bind(this),
     );
+
+    this.messagingSystem.registerActionHandler(
+      `${name}:addNewAccount`,
+      this.addNewAccount.bind(this),
+    );
```

Comment:
I just want to make sure that you're aware that the method `addNewAccount` only adds account to the HD keyring.

Reply:
Hi @gantunesr , thanks for flagging! 
Yes, account syncing will only sync accounts that are derived from the HD keyring, so that's what we want - for now at least 😀